repair/repair.cc

```diff
@@ -2137,10 +2154,13 @@ future<> repair_service::repair_tablets(repair_uniq_id rid, sstring keyspace_nam
                 auto shard = r.shard;
                 auto ip = co_await host2ip(r.host);
                 if (r.host != myhostid) {
-                    rlogger.debug("repair[{}] Repair get neighbors table={}.{} hostid={} shard={} ip={} myip={} myhostid={}",
-                            rid.uuid(), keyspace_name, table_name, r.host, shard, ip, myip, myhostid);
-                    shards.push_back(shard);
-                    nodes.push_back(ip);
+                    bool select = data_centers.empty() ? true : dc_endpoints.contains(ip);
```

Comment:
_addr_map which is used by host2ip is not updated atomically with the topology object, so this is prone to a race on IP change where we can miss the fact that the node which changed IP belongs to the datacenter.
It would be safer to compare host_ids.

Reply:
We could but it is the least thing to consider. We still use ip for most of the things inside repair.